inc/field/fieldsfield.class.php

```diff
@@ -58,20 +54,16 @@ class FieldsField extends PluginFormcreatorAbstractField
    /** @var PluginFieldsField $field */
    public $field = null;
 
-   /**
-    *
-    * @param array $question PluginFormcreatorQuestion instance
-    */
-   public function __construct(PluginFormcreatorQuestion $question) {
-      $this->question  = $question;
-      if (isset($this->question->fields['values'])) {
+   public function getField(): PluginFieldsField {
+      if ($this->field === null && isset($this->question->fields['values'])) {
          $decodedValues = json_decode($this->question->fields['values'], JSON_OBJECT_AS_ARRAY);
          $field_name = $decodedValues['dropdown_fields_field'] ?? '';
          $fieldObj = new PluginFieldsField();
          if ($fieldObj->getFromDBByCrit(['name' => $field_name])) {
-            $this->field  = $fieldObj;
+            $this->field = $fieldObj;
          }
       }
+      return $this->field;
```

Comment:
still a risk to return null here; either we should throw an exception, either we should alow returning null and handle it after each call of this method.